common/djangoapps/enrollment/data.py

```diff
@@ -298,27 +295,51 @@ def get_course_enrollment_info(course_id, include_expired=False):
     else:
         return CourseSerializer(course, include_expired=include_expired).data
 
+
 def get_roster(course_id):
     """
     Returns roster with PII of all enrollees in course
     """
     course_key = CourseKey.from_string(course_id)
-    enrollments = CourseEnrollment.objects.filter(
+
+    # Select appropriate enrollment, user and profile fields for enrolled students.
+    enrollments = CourseEnrollment.objects.select_related(
+        'user'
+    ).select_related(
+        'user__profile'
+    ).filter(
         course_id=course_key,
         is_active=True,
-    ).order_by('user__username')
-
-    roster = [
-        {
-            'anonymous_user_id': anonymous_id_for_user(enrollment.user, None),
-            'user_id': enrollment.user.id,
-            'username': enrollment.user.username,
-            'email': enrollment.user.email,
-            'mode': enrollment.mode,
-            'is_staff': user_has_role(enrollment.user, CourseStaffRole(course_key)),
-            'name': getattr(UserProfile.objects.get(user=enrollment.user), 'name', ''),
-        }
-        for enrollment in enrollments
-    ]
+    ).values(
+        'user__id',
+        'user__username',
+        'user__email',
+        'mode',
+        'user__profile__name',
+    ).order_by(
+        'user__username'
+    )
+
+    # Find all user_ids with instructor or staff roles in course.
+    non_students = CourseAccessRole.objects.filter(
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user_id').distinct()
 
+    roster = []
+    for enrollment in enrollments:
+        is_staff = 0
+        for non_student in non_students:
```

Comment:
@dcadams This is fine as-is, but just as "trivia", here's a more Pythonic approach:

```python
is_staff = any(True for x in non_students if x.values()[0] == enrollment['user_id'])
```